tests/data/test_delta_dataset_source.py

```diff
@@ -113,3 +115,123 @@ def test_delta_dataset_source_too_many_inputs(spark_session, tmp_path):
 
     with pytest.raises(MlflowException, match='Must specify exactly one of "path" or "table_name"'):
         DeltaDatasetSource(path=tmp_path, delta_table_name="temp_delta_too_many_inputs")
+
+
+def test_uc_table_id_retrieval_works(spark_session, tmp_path):
+    def mock_resolve_table_name(table_name, spark):
+        if table_name == "temp_delta_versioned_with_id":
+            return "default.temp_delta_versioned_with_id"
+        return table_name
+
+    def mock_lookup_table_id(table_name):
+        if table_name == "default.temp_delta_versioned_with_id":
+            return "uc_table_id_1"
+        return None
+
+    with mock.patch(
+        "mlflow.data.delta_dataset_source.get_full_name_from_sc",
+        side_effect=mock_resolve_table_name,
+    ), mock.patch(
+        "mlflow.data.delta_dataset_source.DeltaDatasetSource._lookup_table_id",
+        side_effect=mock_lookup_table_id,
+    ), mock.patch(
+        "mlflow.data.delta_dataset_source._get_active_spark_session",
+        return_value=None,
+    ), mock.patch(
+        "mlflow.data.delta_dataset_source.DeltaDatasetSource._is_databricks_uc_table",
+        return_value=True,
+    ):
+        df = pd.DataFrame([[1, 2, 3], [1, 2, 3]], columns=["a", "b", "c"])
+        df_spark = spark_session.createDataFrame(df)
+        df_spark.write.format("delta").mode("overwrite").saveAsTable(
+            "default.temp_delta_versioned_with_id", path=tmp_path
+        )
+
+        df2 = pd.DataFrame([[1, 2, 3]], columns=["a", "b", "c"])
+        df2_spark = spark_session.createDataFrame(df2)
+        df2_spark.write.format("delta").mode("overwrite").saveAsTable(
+            "default.temp_delta_versioned_with_id", path=tmp_path
+        )
+
+        delta_datasource = DeltaDatasetSource(
+            delta_table_name="temp_delta_versioned_with_id", delta_table_version=1
+        )
+        loaded_df_spark = delta_datasource.load()
+        assert loaded_df_spark.count() == df2_spark.count()
+        assert delta_datasource.to_json() == json.dumps(
+            {
+                "delta_table_name": "default.temp_delta_versioned_with_id",
+                "delta_table_version": 1,
+                "is_databricks_uc_table": True,
+                "delta_table_id": "uc_table_id_1",
+            }
+        )
+
+
+@pytest.mark.parametrize(
+    ("call_endpoint_response", "expected_lookup_response"),
+    [
+        (None, None),  # This will test throwing
```

Comment:
(we can use `side_effect` instead of `return_value` in `mock.patch` to make the method throw)

Reply:
Returning None will cause a throw when the table_id is referenced in the response.  I can add the `call_endpoint` itself throwing, but there is no practical difference.

Reply:
> Returning None will cause a throw when the table_id is referenced in the response. I can add the call_endpoint itself throwing, but there is no practical difference.

Sounds good to make `call_endpoint` throw, it's better since it's less subject to changes in how we handle the API response & more-realistically simulates what callers would see in reality

Reply:
Another way to put it: in a typed language, we wouldn't be able to return `None` from `call_endpoint` unless we made the return value an `Option` or returned `null`. Better to match the original method signature (throw or return a response) when mocking it in tests